### PR TITLE
[WIP] Edibles Rework

### PR DESCRIPTION
**Pull request by HenryAhnburger**
#### Summary
SUMMARY: Content "Adds New Marijuana Laced Edibles, replaces WEED_CAKE with WEED_EDIBLE, adds delay to edibles"

#### Purpose of change
The  first purpose of this mess is to add new marijuana laced edibles, all of which are products that exist in the real world.
The second purpose of this PR is to replace `WEED_CAKE` (found in iuse.cpp) with `WEED_EDIBLE`, adding a delay to the high generated from edibles, and allowing for edibles to be added more easily in the future.
Adding this content would bring the game closer to mirroring reality, which is the goal as I understand it.
Switching to `WEED_EDIBLE` will make it easier for others to contribute weed code.

#### Describe the solution
This PR replaces `WEED_CAKE` with `WEED_EDIBLE`. `WEED_EDIBLE` checks if what you're consuming is a food or drink, and prints an appropriate line of text to the player, rather than the hardcoded line from `WEED_CAKE`. It then creates a timed event (`EDIBLE_DIGESTED`) which triggers the high after a (currently hardcoded) 30 minutes.

Canna cookies copy from the regular cookies, and have a copy from recipe.
The other edibles are also copy-froms, but don't have recipes since their generic versions don't have recipes.
They should spawn in dispensaries via the stoner item group, college student mapextras, police evidence, and a few domestic rooms where weed could already spawn.

#### Describe alternatives you've considered
I tried to implement the delayed high by creating a "digesting edible" effect that triggers the regular weed_high effect every time it decays in intensity. (So that ingesting a second edible could just add 1 to the intensity)

This didn't work because there is no "on_decay" option for effects, only the decay message.
I also tried inserting the code to add a player high in the effect code where decay messages are printed, but received a bunch of errors.

I tried a similar route with `effect_int_change`, but it didn't seem like that had a way to differentiate between effects increasing or decreasing in intensity. I tried making a duplicate `on_effect_int_change` with the ability to check for that and cfound that it wasn't always called when I thought it would.

#### Testing
I compared the increased fun values against the increased fun values for space cake and special brownies vs their regular versions.

I compared the post apoc prices to the price for a joint, since most of these items are relatively non-perishable

Items have been test spawned in a mingw 64 bit environment

#### Additional context
This is my first PR, and I'm certain that I have no idea what I'm doing.

This current implementation gets the player character high if they feed an edible to an NPC. I'm trying to find a better implementation to stop this.

### CODE REVIEW

**Comment by HenryAhnburger** [comment]
If (when) it fails the tests, I'll be back tomorrow (US time) to look at it

File: data/json/item_actions.json [hunk 1]
```diff
@@ -969,6 +969,11 @@
     "id": "WEED_CAKE",
     "name": { "str": "Consume" }
   },
+  {
+	"type": "item_action",
```
Comment (Qrox) [inline data/json/item_actions.json:973]: Please lint your changes.

File: data/json/item_actions.json [hunk 2]
```diff
@@ -969,6 +969,11 @@
...
+	"id": "WEED_DRINK",
+	"name": { "str": "Drink" }
```
Comment (Qrox) [inline data/json/item_actions.json:975]:
```suggestion
	"name": { "ctxt": "verb", "str": "Drink" }
```
Some languages have difference words for the noun and verb of "drink", this ensures they can be separately translated.

**Comment by HenryAhnburger** [comment]
I'm assuming those checks are public for this message. I've looked at the travis checks, and I'm not sure I'll be able to parse them myself even tomorrow (US time) with a fresh look. If they're saying something obvious to you, the seasoned github user reading this, **please** lmk.

**Comment by Qrox** [comment]
The failures in the travis build seem related to this PR.

**Comment by anothersimulacrum** [comment]
They don't to me. There's the mingw build failing, that #42870 attempts to treat, and some xcode/macos failure about events not being sent, which I don't think this PR would be effecting.

**Comment by Qrox** [comment]
Yeah, I meant unrelated, but somehow typed related.

**Comment by HenryAhnburger** [comment]
So, what, do I wait and follow #42870 's progress/wait until all the other 12/14 test PRs start going through and then push an arbitrary commit to get the tests to run again?

(and sorry for asking for all the hand-holding, but this issue right now seems fairly non-standard to me)

**Comment by anothersimulacrum** [comment]
Not your fault, don't worry, just wait for someone to either give a review or merge.

**Comment by Fris0uman** [comment]
There might be a way to not hardcode one more iuse for this. Maybe by using the `consum_drug` use action https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_INFO.md#use-actions you can set it all in json. I think that would be better.

**Comment by HenryAhnburger** [comment]
I don't like my lazy solution either. What I'd like to do is replace weed_cake and weed_drink with like weed_edible, and maybe add a delay on the effect of the high, since whoever added weed_cake clearly hasn't had an edible in their life/didn't know how to code either. I'm currently trying to hunt down the delay on the effect of an aspirin so I can copy/paste the delay from there.

**Comment by Night-Pryanik** [comment]
Closing as a WIP PR to clean up open PR list a bit.

**Pull request closed without merging**